src/main/java/net/snowflake/ingest/connection/ServiceResponseHandler.java

```diff
@@ -187,15 +227,26 @@ public static <T> T unmarshallStreamingIngestResponse(
    *
    * @param response HttpResponse
    * @param requestId
+   * @param apiName enum to represent the corresponding api name
+   * @param httpClient HttpClient for retries
+   * @param request HttpRequest for retries
+   * @param requestBuilder RequestBuilder for retries
+   * @return modified HttpResponse
    * @throws IOException if our entity is somehow corrupt or we can't get it
    * @throws IngestResponseException - for all other non OK status
    * @throws BackOffException - if we have a 503 issue
    */
-  private static void handleExceptionalStatus(
-      HttpResponse response, UUID requestId, ApiName apiName)
+  private static HttpResponse handleExceptionalStatus(
+      HttpResponse response,
+      UUID requestId,
+      ApiName apiName,
+      CloseableHttpClient httpClient,
+      HttpUriRequest request,
+      RequestBuilder requestBuilder)
       throws IOException, IngestResponseException, BackOffException {
-    StatusLine statusLine = response.getStatusLine();
-    if (!isStatusOK(statusLine)) {
+    int authRetries = 0;
+    while (!isStatusOK(response.getStatusLine())) {
```

Comment:
is it going in an infinite loop if exception is non SC_UNAUTHORIZED?

Reply:
I think it would throw exception immediately if status is neither 401 nor 503. ([ref](https://github.com/snowflakedb/snowflake-ingest-java/blob/80caea6f0fc85177a374bcd1cd281a8c241559e9/src/main/java/net/snowflake/ingest/connection/ServiceResponseHandler.java#L273))

Reply:
ack!